.github/workflows/style.yaml

```diff
@@ -0,0 +1,71 @@
+# Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
+# Need help debugging build failures? Start at https://github.com/r-lib/actions#where-to-find-help
+on:
+  pull_request:
```

Comment:
It should run on all PRs this way. Which seems right to me, in general. But I'm going to admit this is a pretty heavy-handed way of doing things: I don't know if there is any way to "undo" a styling. Which is probably fine, but potentially annoying.

Reply:
> I don't know if there is any way to "undo" a styling. Which is probably fine, but potentially annoying.

I think this is a good point to bring up. If we ever want to change what's committed, can we make a change and then do something like temporarily disabling the check for those changes? That seems like a potential work around but also kind of annoying. 

> There is also the option to run the workflow not on a PR but triggered by specific comments (like /style, but also /document), which is what they do in ggplot2: https://github.com/tidyverse/ggplot2/blob/main/.github/workflows/pr-commands.yaml

I'm a little confused on how this would actually work? So if you leave a comment on a PR with `/style` it will apply the rules to that section or that file depending on where the comment is? I feel like this could miss things?

Reply:
> I'm a little confused on how this would actually work? So if you leave a comment on a PR with /style it will apply the rules to that section or that file depending on where the comment is? I feel like this could miss things?

It's a bit simpler than that. Somebody leaves a comment in the PR (not a line comment, I don't think) that just says "/style" and the styling action gets run on everything. That then gives the opportunity to revert if there is something we didn't want styled.

Reply:
Ohh I do like that!